web/src/components/search/search.tsx

```diff
@@ -0,0 +1,123 @@
+import { Button, InputGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
+import { AngleDownIcon, AngleUpIcon, SearchIcon, TimesIcon } from '@patternfly/react-icons';
+import _ from 'lodash';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+export type SearchEventType = 'change' | 'searchNext' | 'searchPrevious';
+
+export type SearchEvent = {
+  searchValue: string;
+  type: SearchEventType;
+};
+
+export type SearchHandle = {
+  updateIndicators: (count: string, indicators: ValidatedOptions) => void;
+};
+
+export const SearchComponent: React.FC<{
+  isDark?: boolean;
+  setSearchEvent: (se: SearchEvent) => void;
+  ref?: React.Ref<SearchHandle>;
+  children?: React.ReactNode;
+  // eslint-disable-next-line react/display-name
+}> = React.forwardRef((props, ref: React.Ref<SearchHandle>) => {
```

Comment:
Can you elaborate what's this `forwardRef` / why it's needed?
Also, apparently there's a workaround to not have the linter warning, cf this comment: https://github.com/jsx-eslint/eslint-plugin-react/issues/2269#issuecomment-491567533

Reply:
Forward ref allows you to get [a ref](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R196) from [a child component](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R993):
https://reactjs.org/docs/forwarding-refs.html

This is usefull here to be able to propagate [search ref](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R845-R846) and [expose custom functions](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-b3601c183b04e4c8caaf93d6bdf26a3dc2188ea67c8a9d36cb558bfa9dfbed65R14) to child components.
I assume we can have search features in other tabs so with that approach we will be able to reuse the same search component everywhere.